tests/tests/test_preauth.py

```diff
@@ -222,32 +218,34 @@ def get_pub_key():
         )
 
     @staticmethod
-    def substitute_id_data(id_data_dict):
+    def substitute_id_data(device, id_data_dict):
         """Change the device's identity by substituting it's id data helper script."""
 
         id_data = '#!/bin/sh\n'
         for k,v in id_data_dict.items():
             id_data += 'echo {}={}\n'.format(k,v)
 
         cmd = 'echo "{}" > {}'.format(id_data, Client.ID_HELPER)
-        run(cmd)
+        device.run(cmd)
 
     @staticmethod
-    def restart():
+    def restart(device):
         """Restart the mender service."""
 
-        run('systemctl restart mender.service')
+        device.run("systemctl restart mender.service")
 
     @staticmethod
-    def have_authtoken():
+    def have_authtoken(device):
         """Verify that the device was authenticated by checking its data store for the authtoken."""
         sleepsec = 0
         while sleepsec < Client.MENDER_STORE_TIMEOUT:
             try:
-                out = run('strings {} | grep authtoken'.format(Client.MENDER_STORE))
+                out = device.run(
+                    "strings {} | grep authtoken".format(Client.MENDER_STORE)
+                )
                 return out != ''
             except:
-                output_from_journalctl = run("journalctl -u mender -l")
+                output_from_journalctl = device.run("journalctl -u mender -l")
```

Comment:
Note: When the renaming in `MEN-2948` is merged, this will not work anylonger. So let's be careful to rebase this if it is not merged before.

Reply:
Noted. Let's see who merges first :smile: